src/snowflake/cli/api/project/schemas/native_app/path_mapping.py

```diff
@@ -1,10 +1,43 @@
 from __future__ import annotations
 
-from typing import Optional
+from typing import List, Literal, Optional
 
+from pydantic import Field
 from snowflake.cli.api.project.schemas.updatable_model import UpdatableModel
 
+# Limited to python options only for now.
+ProcessorOptions = Literal["python-snowpark", "PYTHON-SNOWPARK"]
+VirtualEnvTypeOptions = Literal["conda", "CONDA", "venv", "VENV"]
+LanguageOptions = Literal["python", "PYTHON"]
+
+
+class VirtualEnvironment(UpdatableModel):
+    name: str = Field(
+        title="Name of the virtual environment to run the processor in.",
+    )
+    env_type: VirtualEnvTypeOptions = Field(
+        title="Type of the virtual environment to activate.",
+        default="venv",
+    )
+
+
+class AnnotationProcessor(UpdatableModel):
+    language: LanguageOptions = Field(
+        title="Programming language in which the annotated code is written.",
+    )
+    language_version: str = Field(
```

Comment:
Do we need this? IIRC snowpark uses the annotations for this, or defaults to the version of python that we'll be invoking to run the user's code.